src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,268 @@
+import com.ctre.phoenix6.controls.PositionTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private PositionTorqueCurrentFOC leftPositionCurrentRequest;
+  private VelocityTorqueCurrentFOC leftVelocityRequest;
+
+  private PositionTorqueCurrentFOC rightPositionCurrentRequest;
+  private VelocityTorqueCurrentFOC rightVelocityRequest;
+
+  private double leftRequestedVelocity;
+  private double leftRequestedSetpoint; 
+
+  private double rightRequestedVelocity;
+  private double rightRequestedSetpoint;
+
+  private StatusSignal<Double> leftVelocityRPMStatusSignal;
+  private StatusSignal<Double> leftPositionStatusSignal;
+  private StatusSignal<Double> leftStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> leftSetpointStatusSignal;
+  private StatusSignal<Double> leftVoltageRequest;
+
+  private StatusSignal<Double> rightVelocityRPMStatusSignal;
+  private StatusSignal<Double> rightPositionStatusSignal;
+  private StatusSignal<Double> rightStatorCurrentAmpsStatusSignal;
+  private StatusSignal<Double> rightSetpointStatusSignal;
+
+  private Alert configAlert =
+      new Alert("Failed to apply configuration for Climber.", AlertType.ERROR);
+
+  private final TunableNumber leftMotorKP = new TunableNumber("Climber/leftMotorKP", LEFT_POSITION_PID_P);
+  private final TunableNumber leftMotorKI = new TunableNumber("Climber/leftMotorKI", LEFT_POSITION_PID_I);
+  private final TunableNumber leftMotorKD = new TunableNumber("Climber/leftMotorKD", LEFT_POSITION_PID_D);
+  private final TunableNumber leftMotorKS =
+      new TunableNumber("Climber/leftMotorKS", LEFT_POSITION_PID_KS);
+
+  private final TunableNumber rightMotorKP = new TunableNumber("Climber/rightMotorKP", RIGHT_POSITION_PID_P);
+  private final TunableNumber rightMotorKI = new TunableNumber("Climber/rightMotorKI", RIGHT_POSITION_PID_I);
+  private final TunableNumber rightMotorKD = new TunableNumber("Climber/rightMotorKD", RIGHT_POSITION_PID_D);
+  private final TunableNumber rightMotorKS =
+      new TunableNumber("Climber/rightMotorKS", RIGHT_POSITION_PID_KS);
+
+  /** Create a TalonFX-specific generic SubsystemIO */
+  public ClimberIOTalonFX() {
+    leftVelocityRPMStatusSignal = leftMotor.getRotorVelocity();
```

Comment:
probably want to get the rotational velocity of the drum; so, invoke getVelocity and specify the sensor to mechanism ratio to capture the gear ratio